### DISCUSSION TITLE
Applying the lemma detection only to words that have two or more possible parts of speech.

### DISCUSSION THREAD

**Discussion by Vuizur** [xxyzz/WordDumb#247] [Ideas]
In French, when you activate POS detection, words such as participles, for example [glaçant](https://en.wiktionary.org/wiki/glacer#French), don't get annotated with their English translation. This is because glaçant is a participle, but appears in the declination table for the verb "glacer". So spacy detects a participle, but the Worddumb data has it as a verb, which is why it gets no English translation.
My suggestion would be to perform POS detection only for words that have 2 or more possible parts of speech, and handle clear cases such as glaçant like the multi-word-phrases and ignore the grammar.

PS: I have been using the project quite a lot recently, it really works well. Thanks a lot for the active development!

**Comment by xxyzz** [comment]
Please try https://github.com/xxyzz/WordDumb/commit/b8b109284dc806a8afa98148d3d0646513c1f670

**Reply by Vuizur** [reply]
Thanks a lot for the very quick commit. Unfortunately it doesn't seem to work for me yet. For example in this public domain book, the word "véreuses" (a few pages in) is not annotated when using POS detection, but it is when turning it off.
[pg63979-images-3.zip](https://github.com/user-attachments/files/16399069/pg63979-images-3.zip)

**Reply by xxyzz** [reply]
Fixed in 32e45599a8304ff3a5bb543c7cc8409905f1d19d and 9af0c6e917b0e200df700c9569e8d2a0de118867

"use pos" option is removed, the code could find all enabled lemmas and their forms even if spaCy returns wrong POS or wrong lemma.

**Reply by Vuizur** [reply]
Thanks a lot for implementing this! I tested it, and it works great. 👍 I think your solution seems to be really the best.
